equational_theories/Subgraph.lean

```diff
@@ -357,41 +307,64 @@ theorem Equation1571_implies_Equation4656 {G : Type} [Magma G] (h : Equation1571
   fun x y z ↦ ProveEquation1571Consequence (n := 2) h
     {lhs := (Lf 0 ⋆ Lf 1) ⋆ Lf 1, rhs := (Lf 0 ⋆ Lf 2) ⋆ Lf 2} rfl fun | 0 => x | 1 => y | 2 => z
 
-/-- This result was first obtained by Kisielewicz in 1997 via computer assistance. -/
-@[equational_result]
-theorem Equation1689_implies_Equation2 (G: Type*) [Magma G] (h: Equation1689 G) : Equation2 G:=
-  have h9: ∀ a b : G, a ◇ ((a ◇ b) ◇ b) = a :=
-    fun a b ↦
-    calc
-      a ◇ ((a ◇ b) ◇ b) = ((a ◇ a) ◇ ((a ◇ b) ◇ b)) ◇ ((a ◇ b) ◇ b) := by rw [← h a a b]
-      _ = (a ◇ a) ◇ ((a ◇ b) ◇ b) := by rw [Lemma_eq1689_implies_h7 G h]
-      _ = a := by rw [← h]
-  have h10: ∀ a b c : G, a ◇ (a ◇ (b ◇ c)) = a :=
-    fun a b c ↦
-    calc
-      a ◇ (a ◇ (b ◇ c)) = a ◇ ((a ◇ (b ◇ c)) ◇ (b ◇ c)) := by rw [Lemma_eq1689_implies_h7 G h]
-      _ = a := h9 a (b ◇ c)
-  have h11: ∀ a b : G, (a ◇ b) ◇ b = a ◇ b :=
-    fun a b ↦
-    calc
-      (a ◇ b) ◇ b = (a ◇ b) ◇ (((a ◇ b) ◇ ((b ◇ b) ◇ b)) ◇ ((b ◇ b) ◇ b)) := by rw [Lemma_eq1689_implies_h8 G h]
-      _ = a ◇ b := by rw [h9]
-  have h12: ∀ a : G, (a ◇ a) ◇ a = a :=
-    fun a ↦
-    calc
-      (a ◇ a) ◇ a = a ◇ (a ◇ ((a ◇ a) ◇ a)) := by rw [Lemma_eq1689_implies_h4 G h]
-      _ = a := by rw [h10]
-  have h13: ∀ a b : G, (a ◇ b) ◇ b = b :=
-    fun a b ↦
-    calc
-      (a ◇ b) ◇ b = (a ◇ b) ◇ ((b ◇ b) ◇ b) := by rw [h12]
-      _ = b := by rw [← h]
-  have h14: ∀ a b : G, a ◇ b = b := fun _ _ ↦ h11 .. ▸ h13 ..
-  fun a b ↦
-  calc
-    a = a ◇ ((a ◇ b) ◇ b) := by rw [h9 a b]
-    _ = a ◇ b := by rw [h14 (a ◇ b) b]
-    _ = b := h14 a b
+
+/- Proof that 1689 implies 2.  First proven by Kisielewicz in 1997 via computer assistance. -/
+namespace Eq1689
+def pow3 {G: Type*} [Magma G] (a : G) : G := (a ◇ a) ◇ a
+def pow5 {G: Type*} [Magma G] (a : G) : G := ((pow3 a) ◇ a) ◇ a
+def f {G: Type*} [Magma G] (a b : G) : G := (a ◇ b) ◇ b
+def g {G: Type*} [Magma G] (a b : G) : G := a ◇ ((a ◇ b) ◇ b)
+
+-- Rewrite equation in terms of f
+lemma eq_f {G: Type*} [Magma G] (h: Equation1689 G) :
+    ∀ a b c : G, a = (b ◇ a) ◇ (f a c) := by exact h
+
+-- Assuming there always exist b such that f a b = a, finish the proof
+lemma lem_fixf_implies_eq2 {G: Type*} [Magma G] (h: Equation1689 G)
+    (hfixf: ∀ b : G, ∃ c : G, f b c = b) : Equation2 G := by
+  have fproj (a b : G) : f a b = b := by
+    obtain ⟨c, hfbc⟩ := hfixf b
+    nth_rewrite 2 [eq_f h b a c]
+    rw [hfbc] ; rfl
+  have eq30 (a b c : G) : a = (b ◇ a) ◇ c := by
+    rw [← fproj a c, ← eq_f h a b c]
+  have eq6 (a b : G) : a = b ◇ b := by
+    rw [eq30 a (a ◇ b) b, ← eq30 b a a]
+  intro a b ; rw[eq6 a b, ← eq6 b b]
+
+lemma lem_1 {G: Type*} [Magma G] (h: Equation1689 G) :
+    ∀ a b c : G, a ◇ (f (f a b) c) = f a b := by
+  intro a b c
+  nth_rewrite 1 [eq_f h a a b]
```

Comment:
This will suffice 

```suggestion
  nth_rewrite 1 [eq_f h a a]
```

Tested locally so it can be committed directly if you like.